index.html

```diff
@@ -5112,60 +5079,62 @@ <h3 id="playback-AudioBufferSourceNode">
   }
 
   // Render each sample frame in the quantum
-  while (index &lt; output[0].length) {
+  for (let index = 0; index &lt; numberOfFrames; index++) {
+    // Check that currentTime is within allowable range for playback
     if (currentTime &lt; start || currentTime &gt;= stop) {
-      // emit silence for the output frame for the element specified by index
-      renderFrame(output, index, 0);
+      output.push(0); // this sample frame is silent
+      currentTime += dt;
+      continue;
     }
-    else {
-      if (!started) {
-        // Take note that buffer has started playing and get initial playhead position.
-        bufferTime = offset + ((context.currentTime - start) * computedPlaybackRate);
-        started = true;
-      }
 
-      // Handle loop-related calculations
-      if (loop) {
-        // Determine if looped portion has been entered
-        if (!enteredLoop) {
-          if (offset &lt; actualLoopEnd &amp;& bufferTime &gt;= actualLoopStart) {
-            // playback began before or within loop, and playhead is now past loop start
-            enteredLoop = true;
-          }
-          if (offset &gt;= actualLoopEnd &amp;& bufferTime &lt; actualLoopEnd) {
-            // playback began after loop, and playhead is now prior to the loop end
-            enteredLoop = true;
-          }
-        }
+    if (!started) {
+      // Take note that buffer has started playing and get initial playhead position.
+      bufferTime = offset + ((currentTime - start) * computedPlaybackRate);
+      started = true;
+    }
 
-        // Wrap loop iterations as needed
-        if (enteredLoop) {
-          while (bufferTime &gt;= actualLoopEnd) {
-            bufferTime -= actualLoopEnd - actualLoopStart;
-          }
-          while (bufferTime &lt; actualLoopStart) {
-            bufferTime += actualLoopEnd - actualLoopStart;
-          }
+    // Handle loop-related calculations
+    if (loop) {
+      // Determine if looped portion has been entered for the first time
+      if (!enteredLoop) {
+        if (offset &lt; actualLoopEnd &amp;& bufferTime &gt;= actualLoopStart) {
+          // playback began before or within loop, and playhead is now past loop start
+          enteredLoop = true;
+        }
+        if (offset &gt;= actualLoopEnd &amp;& bufferTime &lt; actualLoopEnd) {
+          // playback began after loop, and playhead is now prior to the loop end
+          enteredLoop = true;
         }
       }
 
-      if (bufferTime &gt;= 0 &amp;& bufferTime &lt; buffer.duration) {
-        renderFrame(output, index, playbackSignal(bufferTime));
-      }
-      else {
-        renderFrame(output, index, 0);
+      // Wrap loop iterations as needed
+      if (enteredLoop) {
```

Comment:
Maybe convert this to 
```
} else {
```
to emphasize that this is the alternative case for the if on line 5099.